src/main/java/presenters/console/Input.java

```diff
@@ -90,6 +93,51 @@ public char getSuit() {
         return line.charAt(0);
     }
 
+    /**
+     * Prompt user to select a rank.
+     *
+     * @return a string encoding the selected rank. The returned string will be one of
+     * {"A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K"}
+     */
+    public String getRank() {
+        String[] RANKS = {"A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K"};
+        System.out.print("Pick a rank (one of " + Arrays.toString(RANKS) + "): ");
+        String line = this.inp.nextLine().trim().toUpperCase();
+
+        while (!Arrays.asList(RANKS).contains(line)) {
+            System.out.print("Invalid selection (" + Arrays.toString(RANKS) + "), try again: ");
+            line = this.inp.nextLine().trim().toUpperCase();
+        }
+        return line;
+    }
+
+    /**
+     * Prompt user to select a player.
+     *
+     * @param currPlayer the current player in the game and is not in the list of players displayed to the user.
+     * @param players the list of all the players in the game.
+     * @return a Player object selected by the user. This Player object is in the list of all the players in the game.
+     */
+    public Player getPlayer(Player currPlayer, Player[] players) {
+        HashMap<String, Player> dict = new HashMap<>();
+        ArrayList<String> playerUsernames = new ArrayList<>();
+        for (Player player : players) {
+            if (!player.equals(currPlayer)) {
+                dict.put(player.getUsername(), player);
+                playerUsernames.add(player.getUsername());
+            }
+        }
+        System.out.print("Pick a player (one of " + playerUsernames + "): ");
+        String line = this.inp.nextLine().trim();
+
+        while (!dict.containsKey(line)) {
+            System.out.print("Invalid selection (" + playerUsernames + "), try again: ");
+            line = this.inp.nextLine().trim();
+        }
+
+        return dict.get(line);
+    }
```

Comment:
Do all the actual `Player` objects need to be passed up this far? Is it possible to just pass a list of usernames since that's all that's required to identify a player? This dependency goes from the IO layer to pretty much the entity layer.